src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -169,6 +171,423 @@ public void Equals_DifferentToOtherType()
+            var interval2 = new DateInterval(start2, end, false);
+            Assert.AreNotEqual(interval1, interval2);
+
+            IEqualityComparer<DateInterval> comparer = DateInterval.NormalizingEqualityComparer;
+            Assert.IsFalse(comparer.Equals(interval1, interval2));
+            Assert.AreNotEqual(comparer.GetHashCode(interval1), comparer.GetHashCode(interval2));  // probably
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_DifferentEnd()
+        {
+            LocalDate start = new LocalDate(2000, 1, 1);
+            LocalDate end1 = new LocalDate(2001, 6, 19);
+            LocalDate end2 = new LocalDate(2001, 6, 20);
+            var interval1 = new DateInterval(start, end1, false);
+            var interval2 = new DateInterval(start, end2, false);
+            Assert.AreNotEqual(interval1, interval2);
+
+            IEqualityComparer<DateInterval> comparer = DateInterval.NormalizingEqualityComparer;
+            Assert.IsFalse(comparer.Equals(interval1, interval2));
+            Assert.AreNotEqual(comparer.GetHashCode(interval1), comparer.GetHashCode(interval2));  // probably
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_DifferentInclusivity()
+        {
+            LocalDate start = new LocalDate(2000, 1, 1);
+            LocalDate end = new LocalDate(2001, 6, 19);
+            var interval1 = new DateInterval(start, end, false);
+            var interval2 = new DateInterval(start, end, true);
+            Assert.AreNotEqual(interval1, interval2);
+
+            IEqualityComparer<DateInterval> comparer = DateInterval.NormalizingEqualityComparer;
+            Assert.IsFalse(comparer.Equals(interval1, interval2));
+            Assert.AreNotEqual(comparer.GetHashCode(interval1), comparer.GetHashCode(interval2));  // probably
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_SameRangeButNotEquivalent()
+        {
+            LocalDate start = new LocalDate(2000, 1, 1);
+            LocalDate end1 = new LocalDate(2001, 6, 19);
+            LocalDate end2 = new LocalDate(2001, 6, 20);
+            var interval1 = new DateInterval(start, end1, true);
+            var interval2 = new DateInterval(start, end2, false);
+            Assert.AreEqual(interval1.Length, interval2.Length);
+            Assert.AreNotEqual(interval1, interval2);
+
+            // This is the key difference between this comparer and standard equality.
+            IEqualityComparer<DateInterval> comparer = DateInterval.NormalizingEqualityComparer;
+            Assert.IsTrue(comparer.Equals(interval1, interval2));
+            Assert.AreEqual(comparer.GetHashCode(interval1), comparer.GetHashCode(interval2));
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_Extremes_Equal()
+        {
+            LocalDate start = new LocalDate(-9998, 1, 1);
+            LocalDate end1 = new LocalDate(9999, 12, 30);
```

Comment:
(Of course, this particular case would be `LocalDate.MinValue.PlusDays(-1)`.)

From what I can see, it would only really help this test and `Calendars/SimpleWeekYearRuleTest.cs`, but we could consider it.

I've pulled these out to test constants though, which seems a lot more sensible than what I had before.